internal/graph/dispatch_throttling_check_resolver_test.go

```diff
@@ -2,231 +2,169 @@ package graph
-			resolveCheckDispatchedCounter++
-			return response, nil
-		}).Times(1)
+		req.GetRequestMetadata().DispatchCounter.Store(201)
 
 		ctx := context.Background()
 
 		_, err := dut.ResolveCheck(ctx, req)
+
 		require.NoError(t, err)
-		require.Equal(t, 1, resolveCheckDispatchedCounter)
-		require.False(t, req.GetRequestMetadata().WasThrottled.Load())
 	})
 
-	t.Run("dispatch_should_use_request_threshold_if_available", func(t *testing.T) {
+	t.Run("zero_max_should_interpret_as_default", func(t *testing.T) {
 		ctrl := gomock.NewController(t)
 		defer ctrl.Finish()
 
 		dispatchThrottlingCheckResolverConfig := DispatchThrottlingCheckResolverConfig{
 			// We set timer ticker to 1 hour to avoid it interfering with test
-			Frequency:        1 * time.Hour,
 			DefaultThreshold: 200,
-			MaxThreshold:     210,
+			MaxThreshold:     0,
 		}
-		dut := NewDispatchThrottlingCheckResolver(dispatchThrottlingCheckResolverConfig)
-		defer dut.Close()
+		mockThrottler := mocks.NewMockThrottler(ctrl)
 
-		initialMockResolver := NewMockCheckResolver(ctrl)
-		dut.SetDelegate(initialMockResolver)
+		dut := NewDispatchThrottlingCheckResolver(&dispatchThrottlingCheckResolverConfig, mockThrottler)
+
+		mockCheckResolver := NewMockCheckResolver(ctrl)
+		dut.SetDelegate(mockCheckResolver)
+
+		mockCheckResolver.EXPECT().ResolveCheck(gomock.Any(), gomock.Any()).Times(1)
+		mockThrottler.EXPECT().Throttle(gomock.Any()).Times(0)
 
 		// this is to simulate how many times request has been dispatched
 		// Since this is a small value, we will not expect throttling
 		req := &ResolveCheckRequest{RequestMetadata: NewCheckRequestMetadata(10)}
-		req.GetRequestMetadata().DispatchCounter.Store(203)
-		response := &ResolveCheckResponse{Allowed: true, ResolutionMetadata: &ResolveCheckResponseMetadata{
-			DatastoreQueryCount: 202,
-		}}
-
-		// Here, we count how many times the resolve check is dispatched by the DUT
-		resolveCheckDispatchedCounter := 0
-		initialMockResolver.EXPECT().ResolveCheck(gomock.Any(), req).DoAndReturn(func(ctx context.Context, req *ResolveCheckRequest) (*ResolveCheckResponse, error) {
-			resolveCheckDispatchedCounter++
-			return response, nil
-		}).Times(1)
+		req.GetRequestMetadata().DispatchCounter.Store(190)
 
 		ctx := context.Background()
-		ctx = telemetry.ContextWithDispatchThrottlingThreshold(ctx, uint32(205))
 
 		_, err := dut.ResolveCheck(ctx, req)
+
 		require.NoError(t, err)
-		require.Equal(t, 1, resolveCheckDispatchedCounter)
-		require.False(t, req.GetRequestMetadata().WasThrottled.Load())
 	})
 
-	t.Run("above_threshold_should_be_throttled", func(t *testing.T) {
+	t.Run("dispatch_should_use_request_threshold_if_available", func(t *testing.T) {
 		ctrl := gomock.NewController(t)
 		defer ctrl.Finish()
 
 		dispatchThrottlingCheckResolverConfig := DispatchThrottlingCheckResolverConfig{
-			Frequency:        1 * time.Microsecond,
+			// We set timer ticker to 1 hour to avoid it interfering with test
```

Comment:
i don't understand where this "1 hour" is coming from, is there some implicit math here that i'm not seeing?

Reply:
Nah, just a long enough time that it is predictable. If we had something like 10 microsecond it would be hard to predict in the unit test